esmvaltool/_recipe.py

```diff
@@ -377,8 +377,7 @@ def _dataset_to_file(variable, config_user):
         drs=config_user['drs'])
     if not files and variable.get('derive'):
         variable = copy.deepcopy(variable)
-        variable['short_name'], variable['field'] = get_required(
-            variable['short_name'], variable['field'])[0]
+        variable['short_name'], variable['field'] = get_required(variable)[0]
```

Comment:
Please don't pass the full `variable` dict if you need only a few items from it, that makes it harder to read the code because it is impossible to see what information is used where.

Reply:
For another PR I need the fx variable `sftlf`. For this, I need to call the function
https://github.com/ESMValGroup/ESMValTool/blob/401e702d61838d742dbdbb6ba7db532f83a7145b/esmvaltool/_data_finder.py#L250
which relies on `variable`.

Is there another (easier) way to access fx variables?

Reply:
no, the function that gets the fx variable dictionary is `_get_input_fx_filelist()` but you can just call it once in `_recipe.py` when putting its output in `variable` and then use it in your `get_required()` and make that an optional argument in your function, initialized with `fx_files = None`?

Reply:
Thanks V., I'll try that!